builtin/providers/azurerm/resource_arm_container_registry.go

```diff
@@ -0,0 +1,237 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"net/http"
+
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/arm/containerregistry"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmContainerRegistryCreate,
+		Read:   resourceArmContainerRegistryRead,
+		Update: resourceArmContainerRegistryCreate,
+		Delete: resourceArmContainerRegistryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMContainerRegistryName,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"admin_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"storage_account": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"access_key": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+						},
+					},
+				},
+			},
+
+			"login_server": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"admin_username": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"admin_password": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmContainerRegistryCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).containerRegistryClient
+	log.Printf("[INFO] preparing arguments for AzureRM Container Registry creation.")
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+
+	adminUserEnabled := d.Get("admin_enabled").(bool)
+	tags := d.Get("tags").(map[string]interface{})
+
+	parameters := containerregistry.Registry{
+		Location: &location,
+		RegistryProperties: &containerregistry.RegistryProperties{
+			AdminUserEnabled: &adminUserEnabled,
+		},
+		Tags: expandTags(tags),
+	}
+
+	if v, ok := d.GetOk("storage_account"); ok {
```

Comment:
If this is a required field, we don't need the `if v, ok` part - we can assume it's always there